UACloudLibraryServer/PostgreSQLDB.cs

```diff
@@ -422,5 +427,387 @@ public string[] GetAllNamesAndNodesets()
+                        sqlCommand.Parameters.AddWithValue("description", category.Description);
+                        sqlCommand.Parameters.AddWithValue("logourl", category.IconUrl);
+                        sqlCommand.Parameters.AddWithValue("lastmodification", category.LastModificationTime);
+                        sqlCommand.Parameters.AddWithValue("id", category_id);
+                        sqlCommand.ExecuteNonQuery();
+                        sqlCommand.Parameters.Clear();
+                    }
+                }
+            }
+        }
+
+        private void OverwriteOrganisation(Organisation organisation, long organisation_id, NpgsqlCommand sqlCommand)
+        {
+            string sqlQuery = string.Format("SELECT name, contactemail, website, description, logourl FROM organisation WHERE contributor_id = {0}", organisation_id);
+            sqlCommand.CommandText = sqlQuery;
+            using (NpgsqlDataReader reader = sqlCommand.ExecuteReader())
+            {
+                if (reader.Read())
+                {
+                    string name = reader.GetString("name");
+                    string contactemail = reader.GetString("contactemail");
+                    string website = reader.GetString("website");
+                    string description = reader.GetString("description");
+                    string logourl = reader.GetString("logourl");
+
+                    if (name != organisation.Name
+                        || contactemail != organisation.ContactEmail
+                        || website != organisation.Website.ToString()
+                        || description != organisation.Description
+                        || logourl != organisation.LogoUrl.ToString())
+                    {
+                        sqlQuery = "UPDATE organisation SET name = @name, contactemail = @mail, website = @website, description = @description, logourl = @logourl WHERE contributor_id = @id";
+                        sqlCommand.CommandText = sqlQuery;
+                        sqlCommand.Parameters.AddWithValue("name", organisation.Name);
+                        sqlCommand.Parameters.AddWithValue("mail", organisation.ContactEmail);
+                        sqlCommand.Parameters.AddWithValue("website", organisation.Website);
+                        sqlCommand.Parameters.AddWithValue("description", organisation.Description);
+                        sqlCommand.Parameters.AddWithValue("logourl", organisation.LogoUrl);
+                        sqlCommand.Parameters.AddWithValue("lastmodification", organisation.LastModificationTime);
+                        sqlCommand.Parameters.AddWithValue("id", organisation_id);
+                        sqlCommand.ExecuteNonQuery();
+                        sqlCommand.Parameters.Clear();
+                    }
+                }
+            }
+        }
+
+        #endregion
+
+        #region MigrationTool
+        public bool TryMigrateTables()

```

Comment:
How and when to call TryMigrateTables?
How can we make sure that the migration is not executed twice?

Can we persist a storage version or similar on the NodeSet so that it can be recorded in the database whether a NodeSet has already been migrated?